### DISCUSSION TITLE
V7 - Accessing filtered rows post filtering

### DISCUSSION THREAD

**Discussion by JohnCos247** [TanStack/table#2765] [Q&A]
Hey everyone. Does anyone know of a way to access the rows which are filtered from the global and column filters before the re-render of the Table? I am trying to figure out a way to pass the filtered rows back to the parent component.

```
    const count = preGlobalFilteredRows.length;
    const [value, setValue] = React.useState(globalFilter);
    const onChange = useAsyncDebounce(value => {
        
        setGlobalFilter(value || undefined);
        // Do something with the filtered rows here.

    }, 200);
    return (
        <span>
            <input
                value={value || ""}
                onChange={e => {
                    setValue(e.target.value);
                    onChange(e.target.value);
                }}
                placeholder={`${count} records...`}
            />
        </span>
    )
```

**Comment by gargroh** [comment]
This data can be helpful to you https://github.com/tannerlinsley/react-table/blob/master/src/plugin-hooks/useFilters.js#L290 https://github.com/tannerlinsley/react-table/blob/master/src/plugin-hooks/useGlobalFilter.js#L167
